src/gui/maptools/qgsmaptooleditblanksegments.cpp

```diff
@@ -0,0 +1,947 @@
+/***************************************************************************
+    qgsmaptooleditblanksegments.cpp
+    ---------------------
+    begin                : 2025/08/19
+    copyright            : (C) 2025 by Julien Cabieces
+    email                : julien dot cabieces at oslandia dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgscoordinatetransform.h"
+#include "qgsfeatureid.h"
+#include "qgsmaptooleditblanksegments.h"
+#include "moc_qgsmaptooleditblanksegments.cpp"
+
+#include "line_p.h"
+#include "qgsmapcanvas.h"
+#include "qgsvectorlayer.h"
+#include "qgslinesymbollayer.h"
+#include "qgssinglesymbolrenderer.h"
+#include "qgssymbol.h"
+#include "qgsnullpainterdevice.h"
+#include "qgsmapmouseevent.h"
+#include "qgsrubberband.h"
+#include "qgssettingsregistrycore.h"
+#include "qgssettingsentryimpl.h"
+#include "qgsguiutils.h"
+#include "qgssnappingutils.h"
+#include "qgsstyleentityvisitor.h"
+
+constexpr int TOLERANCE = 20;
+
+
+/////////
+
+///@cond PRIVATE
+
+namespace
+{
+
+  class SymbolLayerVisitor : public QgsStyleEntityVisitorInterface
+  {
+    public:
+      //! constructor
+      SymbolLayerVisitor( const QString &symbolLayerId )
+        : mSymbolLayerId( symbolLayerId ) {}
+
+      bool visitEnter( const QgsStyleEntityVisitorInterface::Node &node ) override
+      {
+        if ( node.type != QgsStyleEntityVisitorInterface::NodeType::SymbolRule )
+          return false;
+
+        return true;
+      }
+
+      bool visitSymbol( const QgsSymbol *symbol, const QString &leafIdentifier )
+      {
+        if ( symbol && !mSymbol )
+          mSymbol = symbol;
+
+        for ( int idx = 0; idx < symbol->symbolLayerCount(); idx++ )
+        {
+          const QgsSymbolLayer *sl = symbol->symbolLayer( idx );
+          if ( sl->id() == mSymbolLayerId )
+          {
+            mSymbolLayer = dynamic_cast<const QgsTemplatedLineSymbolLayerBase *>( sl );
+            mSymbolLayerIndex = idx;
```

Comment:
what if the cast failed? you will have a null mSymbolLayer but a defined mSymbolLayerIndex?